lc0/src/neural/network_opencl.cc

```diff
@@ -0,0 +1,118 @@
+/*
+    This file is part of Leela Chess Zero.
+    Copyright (C) 2017 Gian-Carlo Pascutto
+    Copyright (C) 2018 The LCZero Authors
+
+    Leela Zero is free software: you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation, either version 3 of the License, or
+    (at your option) any later version.
+
+    Leela Zero is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License
+    along with Leela Zero.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include "network_blas_cl_common.h" // factory.h, network.h, optionsdict.h
+#include "opencl/OpenCLScheduler.h"
+
+namespace lczero {
+
+class OpenCLNetwork : public BlasCLNetwork {
+ public:
+  OpenCLNetwork(const Weights& weights, const OptionsDict& options)
+    : BlasCLNetwork(weights, options) {
+    initialize();
+  }
+
+ protected:
+  void initialize(void);
+  std::vector<float> zeropad_U(const std::vector<float>& U, const int outputs, const int channels, const int outputs_pad, const int channels_pad);
```

Comment:
If doesn't need access to member variable, consider making static or put outside a class into anonymous namespace.